test/functional/mempool_dust.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test dust limit mempool policy (`-dustrelayfee` parameter)"""
+from decimal import Decimal
+
+from test_framework.key import ECKey
+from test_framework.messages import (
+    COIN,
+    CTxOut,
+)
+from test_framework.script import (
+    CScript,
+    OP_RETURN,
+    OP_TRUE,
+)
+from test_framework.script_util import (
+    key_to_p2pk_script,
+    key_to_p2pkh_script,
+    key_to_p2wpkh_script,
+    keys_to_multisig_script,
+    output_key_to_p2tr_script,
+    program_to_witness_script,
+    script_to_p2sh_script,
+    script_to_p2wsh_script,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.test_node import TestNode
+from test_framework.util import (
+    assert_equal,
+    get_fee,
+)
+from test_framework.wallet import MiniWallet
+
+
+DUST_RELAY_TX_FEE = 3000  # default setting [sat/kvB]
+
+
+class DustRelayFeeTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def test_dust_output(self, node: TestNode, dust_relay_fee: Decimal,
+                         output_script: CScript, type_desc: str) -> None:
+        # determine dust threshold (see `GetDustThreshold`)
+        if output_script[0] == OP_RETURN:
+            dust_threshold = 0
+        else:
+            tx_size = len(CTxOut(nValue=0, scriptPubKey=output_script).serialize())
+            tx_size += 67 if output_script.IsWitnessProgram() else 148
+            dust_threshold = int(get_fee(tx_size, dust_relay_fee) * COIN)
+        self.log.info(f"-> Test {type_desc} output (size {len(output_script)}, limit {dust_threshold})")
+
+        # amount right on the dust threshold should pass
+        tx = self.wallet.create_self_transfer()["tx"]
```

Comment:
Not actually needed, but just to make sure ...
```suggestion
        tx = self.wallet.create_self_transfer()["tx"]
        # make sure there's enough value available for the new output we're creating
        tx.vout[0].nValue -= dust_threshold
```
Note that since this tx is spending coinbase, there's almost 50BTC available in the first output, and also spending coinbase doesn't require a signature here in the functional tests, so we can make this adjustment without having to re-signing the transaction.

Reply:
Thanks, done. In fact after extending the dust fee-rate test list with a larger value (333333), the test would fail without this change, so it was very helpful. Note that not requiring a signature is not only true for the coinbase outputs we spend here, but also to outputs created by MiniWallet going to the exact same address.